source/MapDetailPanel.cpp

```diff
@@ -534,6 +533,40 @@ void MapDetailPanel::DrawKey()
 		"Danger level:"
 	};
 	const string &header = HEADER[-min(0, max(-7, commodity))];
+	const vector<Trade::Commodity> &commodities = GameData::Commodities();
+
+	int headerWidth = font.Width(header);
+	int maxWidth = headerWidth;
+	if(commodity >= 0)
+	{
+		const auto &range = commodities[commodity];
+		if(static_cast<unsigned>(commodity) >= commodities.size())
+			return;
+
+		for(int i = 0; i <= 3; ++i)
+		{
+			int price = range.low + ((range.high - range.low) * i) / 3;
+			maxWidth = max(maxWidth, font.Width(to_string(price)));
+		}
+	}
+
+    int longestCommodityWidth = 0;
+    for(const auto &commodity : commodities)
+    {
+        int commodityWidth = font.Width(commodity.name);
+        longestCommodityWidth = max(longestCommodityWidth, commodityWidth);
+    }
+
+    int longestHeaderWidth = 0;
+    for(const auto &header : HEADER)
+    {
+        int headerWidth = font.Width(header);
+        longestHeaderWidth = max(longestHeaderWidth, headerWidth);
+    }
+
+    maxWidth = std::max(maxWidth, std::max(longestCommodityWidth, longestHeaderWidth));
```

Comment:
The project is tab-spaced indenting.  I recommend using an auto-formatter for example https://github.com/samrocketman/endless-sky-vscode-devcontainer/blob/6608e93d4d1e6ce7410eea6ba7e86338df1ec9b6/endless-sky/.vscode/tasks.json#L25-L29